### PR TITLE
[FLINK-11451][table] move *QueryConfig and TableDescriptor to flink-table-api-java

### PR DESCRIPTION
**Pull request by xueyumusic**
## What is the purpose of the change
This PR moves QueryConfig, BatchQueryConfig, StreamQueryConfig and TableDescriptor to flink-table-api-java

## Brief change log

  - *move QueryConfig, BatchQueryConfig, StreamQueryConfig and TableDescriptor to flink-table-java*
  - *remove queryConfig.scala and TableDescriptor.scala*

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (JavaDocs)

### CODE REVIEW

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/QueryConfig.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.io.Serializable;
+
+/**
+ * The {@link QueryConfig} holds parameters to configure the behavior of queries.
+ */
+@PublicEvolving
+public class QueryConfig implements Serializable {
```
Comment (dawidwys) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/QueryConfig.java]: This should be an interface.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/BatchQueryConfig.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package org.apache.flink.table.api;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * The {@link BatchQueryConfig} holds parameters to configure the behavior of batch queries.
+ */
+@PublicEvolving
```
Comment (dawidwys) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/BatchQueryConfig.java]: I wouldn't add this annotation. This class will actually be dropped in a near future, once we have one unified `TableEnvironment`.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/StreamQueryConfig.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.time.Time;
+
+/**
+ * The {@link StreamQueryConfig} holds parameters to configure the behavior of streaming queries.
+ *
+ * <p>An empty {@link StreamQueryConfig} can be generated using the [[StreamTableEnvironment.queryConfig]]
+ * method.
+ */
+@PublicEvolving
```
Comment (dawidwys) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/StreamQueryConfig.java]: Same as for `BatchQueryConfig`. Let's not add this annotation.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/StreamQueryConfig.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+package org.apache.flink.table.api;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.time.Time;
+
+/**
+ * The {@link StreamQueryConfig} holds parameters to configure the behavior of streaming queries.
+ *
+ * <p>An empty {@link StreamQueryConfig} can be generated using the [[StreamTableEnvironment.queryConfig]]
```
Comment (dawidwys) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/StreamQueryConfig.java]: Please convert `[[StreamTableEnvironment.queryConfig]]` as well.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/StreamQueryConfig.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+			throw new IllegalArgumentException(
+					"Difference between minTime: " + minTime.toString() + " and maxTime: " + maxTime.toString() +
+					"shoud be at least 5 minutes.");
+		}
+		minIdleStateRetentionTime = minTime.toMilliseconds();
+		maxIdleStateRetentionTime = maxTime.toMilliseconds();
+		return this;
+	}
+
+	public long getMinIdleStateRetentionTime() {
```
Comment (dawidwys) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/StreamQueryConfig.java]: Could you add javadoc for those methods? I think we can reuse the description for corresponding field.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/descriptors/TableDescriptor.java [hunk 6]
```diff
@@ -16,9 +16,14 @@
...
+package org.apache.flink.table.descriptors;
+
+import org.apache.flink.annotation.PublicEvolving;
 
 /**
   * Common class for all descriptors describing table sources and sinks.
   */
-abstract class TableDescriptor extends DescriptorBase
+@PublicEvolving
+public abstract class TableDescriptor extends DescriptorBase {
```
Comment (dawidwys) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/descriptors/TableDescriptor.java]: Just a thought. How about we convert it to interface? That means converting `DescriptorBase` to interface as well. What do you think @twalthr about it?

Comment (xueyumusic) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/descriptors/TableDescriptor.java]: Hi, @dawidwys , thanks for review. I updated the code except this place right now. 
It looks `DescriptorBase` overrides `toString()` method implementation, if it is converted to interface how to migrate this method...

**Comment by dawidwys** [comment]
Thank you @xueyumusic for your contribution

**Pull request closed without merging**